src/gui/integration/integration_test.go

```diff
@@ -1796,3 +1797,150 @@ func TestLivePendingTransactions(t *testing.T) {
 	_, err := c.PendingTransactions()
 	require.NoError(t, err)
 }
+
+func TestLiveWalletSpend(t *testing.T) {
+	if !doLive(t) {
+		return
+	}
+
+	w, totalCoins, _ := prepareAndCheckWallet(t, 2, 1)
+	tt := []struct {
+		name    string
+		to      string
+		coins   uint64
+		checkTx func(t *testing.T, tx *visor.TransactionResult)
+	}{
+		{
+			"Send all coins to the first address",
+			w.Entries[0].Address.String(),
+			totalCoins,
+			func(t *testing.T, tx *visor.TransactionResult) {
+				// Confirms the total output coins are equal to the totalCoins
+				var coins uint64
+				for _, o := range tx.Transaction.Out {
+					c, err := droplet.FromString(o.Coins)
+					require.NoError(t, err)
+					coins += c
+				}
+				require.Equal(t, totalCoins, coins)
+
+				// Confirms the address balance are equal to the totoalCoins
+				coins, _ = getAddressBalance(t, w.Entries[0].Address.String())
+				require.Equal(t, totalCoins, coins)
+			},
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.name, func(t *testing.T) {
+			c := gui.NewClient(nodeAddress())
```

Comment:
why don't we declare this out of the `for` scope and then use that single instance at other places?